sql/core/src/test/resources/sql-tests/inputs/union.sql

```diff
@@ -9,7 +9,7 @@ FROM   (SELECT * FROM t1
 
 -- Type Coerced Union
 SELECT *
-FROM   (SELECT * FROM t1
+FROM   (SELECT * FROM t1 where c1 = 1
```

Comment:
Great, otherwise a flaky ut will appear, as follows:
<img width="804" alt="image" src="https://github.com/user-attachments/assets/2c8a6fbb-da37-4936-9517-9a26edb7e47b">
https://github.com/panbingkun/spark/actions/runs/11887199373/job/33119593602

Reply:
Thank you, @panbingkun. Some of the other tests that assert task-side failures might also be flaky.